query/src/test/scala/filodb/query/exec/LastSampleFunctionSpec.scala

```diff
@@ -50,7 +50,7 @@ class LastSampleFunctionSpec extends RawDataWindowingSpec {
       val start = now + ThreadLocalRandom.current().nextLong(80000)
       val end = start
       val step = 1
-      val lastSamplesIter = new SlidingWindowIterator(rv.rows, start, step, end, 0, LastSampleFunction, queryConfig)
+      val lastSamplesIter = new SlidingWindowIterator(rv.rows, start, step, end, queryConfig.staleSampleAfterMs, LastSampleFunction, queryConfig)
```

Comment:
what if we do 0 window check in SlidingWindowIterator? would it be more encapsulated/logical?

Reply:
Lets keep consistent with ChunkedWindowIterator